test/parallel/test-runner-no-isolation-different-cwd.mjs

```diff
@@ -0,0 +1,44 @@
+import { allowGlobals, mustCall } from '../common/index.mjs';
+import * as fixtures from '../common/fixtures.mjs';
+import { deepStrictEqual } from 'node:assert';
+import { run } from 'node:test';
+
+const stream = run({
+  cwd: fixtures.path('test-runner', 'no-isolation'),
+  isolation: 'none',
+});
+
+
+stream.on('test:pass', mustCall(5));
+// eslint-disable-next-line no-unused-vars
+for await (const _ of stream);
+allowGlobals(globalThis.GLOBAL_ORDER);
+deepStrictEqual(globalThis.GLOBAL_ORDER, [
+  'before one: <root>',
+  'suite one',
+  'before two: <root>',
+  'suite two',
+
+  'beforeEach one: suite one - test',
+  'beforeEach two: suite one - test',
+  'suite one - test',
+  'afterEach one: suite one - test',
+  'afterEach two: suite one - test',
+
+  'beforeEach one: test one',
+  'beforeEach two: test one',
+  'test one',
+  'afterEach one: test one',
+  'afterEach two: test one',
```

Comment:
To save you some debugging time - I think this will probably fail when you rebase.